### PR TITLE
added tests for wp_nonce_field and wp_referer_field

### PR DESCRIPTION
**Pull request by pbearne**
Trac ticket: https://core.trac.wordpress.org/ticket/55578

### CODE REVIEW

File: tests/phpunit/tests/functions/wpNonceField.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	 * @ticket 55578
+	 */
+	public function data_wp_nonce_field() {
+
+		return array(
+			'default'     => array(
+				- 1,
+				'_wpnonce',
+				true,
+				'#^<input type="hidden" id="_wpnonce" name="_wpnonce" value=".{10}" /><input type="hidden" name="_wp_http_referer" value="" />$#',
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpNonceField.php]:
```suggestion
				'action'           => - 1,
				'name'             => '_wpnonce',
				'referer'          => true,
				'expected_reg_exp' => '#^<input type="hidden" id="_wpnonce" name="_wpnonce" value=".{10}" /><input type="hidden" name="_wp_http_referer" value="" />$#',
```
Adding keys for each parameter in the datasets makes it easier for others to add datasets in the future.

File: tests/phpunit/tests/functions/wpNonceField.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+			),
+			'nonce_name'  => array(
+				- 1,
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpNonceField.php]: Can we change `- 1` to `-1`, i.e. without the space?

File: tests/phpunit/tests/functions/wpRefererField.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+class Tests_Functions_wpRefererField extends WP_UnitTestCase {
+
+	/**
+	 * @ticket 55578
+	 */
+	public function test_wp_referer_field() {
+
+		$_SERVER['REQUEST_URI'] = '/test/';
+		wp_referer_field();
+		$this->expectOutputRegex( '#^<input type="hidden" name="_wp_http_referer" value="/test/" />$#' );
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpRefererField.php]: As this doesn't contain any regular expression syntax except the start and end of a line, `expectOutputString()` can be used instead.

```php
wp_referer_field();

$this->expectOutputString( '<input type="hidden" name="_wp_http_referer" value="/test/" />' );
```

File: tests/phpunit/tests/functions/wpRefererField.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	}
+
+	/**
+	 * @ticket 55578
+	 */
+	public function test_wp_referer_field_return() {
+
+		$_SERVER['REQUEST_URI'] = '/test/';
+
+		$this->assertMatchesRegularExpression( '#^<input type="hidden" name="_wp_http_referer" value="/test/" />$#', wp_referer_field( false ) );
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpRefererField.php]: As this doesn't contain any regular expression syntax except the start and end of a line, `assertSame()` can be used instead.

```php
$this->assertSame( '<input type="hidden" name="_wp_http_referer" value="/test/" />', wp_referer_field( false ) );
```

File: tests/phpunit/tests/functions/wpNonceField.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+	public function test_wp_nonce_field() {
+
+		wp_nonce_field();
+		$this->expectOutputRegex( '#^<input type="hidden" id="_wpnonce" name="_wpnonce" value=".{10}" /><input type="hidden" name="_wp_http_referer" value="" />$#' );
+	}
+
+	/**
+	 * @ticket 55578
+	 *
+	 * @dataProvider data_wp_nonce_field
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpNonceField.php:25]: The `action`, `name`, `referer` and `expected_regexp` parameters should be documented like so:

```php
* @param int|string $action           Action name.
* @param string     $name             Nonce name.
* @param bool       $referer          Whether to set the referer field fior validation.
* @param string     $expected_reg_exp The expected regular expression.
```

File: tests/phpunit/tests/functions/wpNonceField.php [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+				true,
+				'#^<input type="hidden" id="_wpnonce" name="_wpnonce" value="' . wp_create_nonce( 'action_name' ) . '" /><input type="hidden" name="_wp_http_referer" value="" />$#',
+			),
+			'no_referer'  => array(
+				- 1,
+				'_wpnonce',
+				false,
+				'#^<input type="hidden" id="_wpnonce" name="_wpnonce" value=".{10}" />$#',
+			),
+		);
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpNonceField.php:75]: Can we add a dataset that makes sure `$name` always goes through `esc_attr()`?

For example, a `$name` of `a&b` should become `a&amp;b`

File: tests/phpunit/tests/functions/wpNonceField.php [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+	 *
+	 * @dataProvider data_wp_nonce_field
+	 */
+	public function test_wp_nonce_field_return( $action, $name, $referer, $expected_reg_exp ) {
+
+		$this->assertMatchesRegularExpression( $expected_reg_exp, wp_nonce_field( $action, $name, $referer, false ) );
+	}
+
+	/**
+	 * @ticket 55578
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpNonceField.php]:
```suggestion
	 * Data provider.
	 *
	 * @return array
```
The `@ticket` annotation isn't needed for data providers.

**Review by costdev — CHANGES_REQUESTED** [review]
Thanks for the tests Paul! I have suggested some changes in this review.

Will you have time to review this before RC1 on Tuesday 3rd May?

File: tests/phpunit/tests/functions/wpNonceField.php (hunk 6 above)
Comment (pbearne) [reply tests/phpunit/tests/functions/wpNonceField.php:75]: added

**Comment by audrasjb** [comment]
Committed in https://core.trac.wordpress.org/changeset/54420

**Pull request closed without merging**